### PR TITLE
Factory plugin

### PR DESCRIPTION
**Pull request by sunix**
### What does this PR do?
- Removes the legacy Che-Theia factory extension.
- Includes the che-theia-factory plugin.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/11249

#### Release Notes
Che-theia clones projects that are defined in the workspace definition.
Che-theia performs factory cloning and execute post loading actions.

### CODE REVIEW

**Review by AndrienkoAleksandr — APPROVED** [review]

File: dockerfiles/theia/Dockerfile [hunk 1]
```diff
@@ -92,7 +92,7 @@ ADD src/builder/setup-nodejs.sh /tmp/builder/scripts/setup-nodejs.sh
 # Install supervisor for running theia
 # Install which tool in order to search git
 RUN yum install -y centos-release-scl
-RUN yum install -y rh-git29 scl-utils sudo java-1.8.0-openjdk-devel bzip2 which && \
+RUN yum install -y rh-git29 scl-utils sudo java-1.8.0-openjdk-devel bzip2 which make gcc-c++ && \
```
Comment (benoitf) [inline dockerfiles/theia/Dockerfile]: the theia-ide image is only a runtime image so make, gcc and c++ should not be there

**Review by benoitf — CHANGES_REQUESTED** [review]
theia-ide image should not contains any compilation stuff.
(probably it may go in https://github.com/ws-skeleton/che-plugin-dev-tooling which is the "development image for plug-ins, etc) which is in che7 stack

**Comment by sunix** [comment]
@benoitf ok but
at the same time
- it is the image where we do self hosting
- the way we build in docker should be the closest to the way we build in our dev environment.
- why do we include node_modules in /node_modules in this case ?
- why do we have jdk stuff in this image ?

My point is that we are adding restriction that are not so important to me. We should give the choice to the developer to build image anywhere and if for some reason developer prefers to build in che-theia image, why restricting him ?

**Comment by sunix** [comment]
@benoitf I have updated the PR without make and gcc-c++

**Review by benoitf — APPROVED** [review]

**Comment by riuvshin** [comment]
@sunix no `ci-build` / `ci-test` ??? this broke master :)

**Pull request merged**